tools/data-api-sdk/v1/helpers/golang_type_for_sdk_object_definition.go

```diff
@@ -0,0 +1,98 @@
+package helpers
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+
+	"github.com/hashicorp/pandora/tools/data-api-sdk/v1/models"
+)
+
+// GolangTypeForSDKObjectDefinition returns the Golang type name for the SDKObjectDefinition provided in `input`.
+func GolangTypeForSDKObjectDefinition(input models.SDKObjectDefinition, golangPackageName *string) (*string, error) {
+	// TODO: we should look to add unit tests for this method
+
+	// NOTE: all of this validation should be done in the Importer and the API - this is purely sanity checking
+
+	if input.Type == models.CSVSDKObjectDefinitionType {
+		if input.NestedItem == nil {
+			return nil, fmt.Errorf("a csv type must have a nested item but didn't get one")
+		}
+
+		// TODO: determine how to handle these
+		// for now treat CSV's as a string
+		return pointer.To("string"), nil
+	}
+
+	if input.Type == models.DictionarySDKObjectDefinitionType {
+		if input.NestedItem == nil {
+			return nil, fmt.Errorf("a dictionary type must have a nested item but didn't get one")
+		}
+
+		innerType, err := GolangTypeForSDKObjectDefinition(*input.NestedItem, golangPackageName)
+		if err != nil {
+			return nil, fmt.Errorf("determining inner type for dictionary: %+v", err)
+		}
+
+		return pointer.To(fmt.Sprintf("map[string]%s", *innerType)), nil
+	}
+
+	if input.Type == models.ListSDKObjectDefinitionType {
+		if input.NestedItem == nil {
+			return nil, fmt.Errorf("a list type must have a nested item but didn't get one")
+		}
+
+		innerType, err := GolangTypeForSDKObjectDefinition(*input.NestedItem, golangPackageName)
+		if err != nil {
+			return nil, fmt.Errorf("determining inner type for list: %+v", err)
+		}
+
+		return pointer.To(fmt.Sprintf("[]%s", *innerType)), nil
+	}
+
+	if input.Type == models.ReferenceSDKObjectDefinitionType {
+		if input.ReferenceName == nil {
+			return nil, fmt.Errorf("missing Reference for a Reference ObjectDefinition")
+		}
+
+		out := *input.ReferenceName
+		if golangPackageName != nil {
+			out = fmt.Sprintf("%s.%s", *golangPackageName, out)
+		}
+		return pointer.To(out), nil
+	}
+
+	sdkObjectDefinitionTypesToValues := map[models.SDKObjectDefinitionType]string{
+		// Simple Types
+		models.BooleanSDKObjectDefinitionType:  "bool",
+		models.DateTimeSDKObjectDefinitionType: "string", // intentional since we have cast methods one way or the other
+		models.FloatSDKObjectDefinitionType:    "float64",
+		models.IntegerSDKObjectDefinitionType:  "int64",
+		models.StringSDKObjectDefinitionType:   "string",
+
+		// Complex Types
+		models.LocationSDKObjectDefinitionType:  "string",
+		models.RawFileSDKObjectDefinitionType:   "[]byte",
+		models.RawObjectSDKObjectDefinitionType: "interface{}", // TODO: should we standardise this on `any`?
```

Comment:
As of Go 1.18 `any` and `interface{}` are interchangeable (core) types - so this would be mostly a cosmetic change fwiw.

I do think there's _some_ benefit to doing this long term (hence the TODO, since I'd argue that `any` is clearer than `interface{}` to folks less familiar with Go), but this is pretty low priority/isn't urgent by any means (since it'd trigger an SDK regeneration when it does happen, so would need to be done once the refactor is completed at any rate).